pkgs/build-support/fetchurl/default.nix

```diff
@@ -124,6 +124,8 @@ if (!hasHash) then throw "Specify hash for fetchurl fixed-output derivation: ${s
 
   inherit curlOpts showURLs mirrorsFile impureEnvVars postFetch downloadToTemp executable;
 
+  userAgent = "Nix ${builtins.nixVersion} on nixpkgs ${stdenv.lib.nixpkgsVersion}";
```

Comment:
What if we tracked a version specifically for fetchurl, which changes way less than nixpkgs?